src/Cli/dotnet/Commands/Test/VSTest/TestCommand.cs

```diff
@@ -298,11 +298,16 @@ internal static int RunArtifactPostProcessingIfNeeded(string testSessionCorrelat
 
     private static bool ContainsBuiltTestSources(string[] args)
     {
-        foreach (string arg in args)
+        for (int i = 0; i < args.Length; i++)
         {
-            if (!arg.StartsWith("-") &&
```

Comment:
We shouldn't have removed this `StartsWith` check. This is likely the cause of https://github.com/dotnet/sdk/issues/51249.